basic-algorithms.js

```diff
@@ -1,7 +1,81 @@
 // Names and Input
 
+/* 1.*/var hacker1 = "Griffith";
+/* 2.*/console.log("The driver's name is " + hacker1);
+/* 3.*/var hacker2 = prompt("What is your name?");
+/* 4.*/console.log("The navigator's name is " + hacker2); 
 
 //Conditionals
 
+/* 5.*/if(hacker1.length > hacker2.length) {
+    console.log("The Driver has the longest name, it has " + hacker1.length + " charachters");
+} else if(hacker2.length > hacker1.length) {
+    console.log("The Navigator has the longest name, it has " + hacker2.length + " charachters");
+} else {
+    console.log("Both the Navigator and the Driver have the same name length, " + hacker1.length + " characters, wow!")
+}
+
+//Loops
+/* 6.*/for(let i = 0; i < hacker1.length; i++) {
+    var character = (hacker1.charAt(i));
+    console.log(character.toUpperCase() + " ");
+}
+/* 7.*/for(let i = hacker2.length - 1; i > -1; i--) {
+    console.log(hacker2.charAt(i));
+}
+/* 8.*/if (hacker1.localeCompare(hacker2) === -1) {
+    console.log("The driver's name goes first");
+} else if (hacker2.localeCompare(hacker1) === -1) {
+    console.log("Yo, the navigator goes first definitely");
+} else {
+    console.log("you both got the same name bro");
+}
+
+//Palindrome 
+
+/* 9.*/var possiblePalindrome = prompt("See if your word is a palindrome");
+var wordNoSymbols = "";
+for(let i = 0; i < possiblePalindrome; i++) {
+    if(possiblePalindrome.charAt(i) != "," || ossiblePalindrome.charAt(i) != " ") {
+        wordNoSymbols += possiblePalindrome.charAt(i);
+    }
+}
+
+var firstHalf = wordNoSymbols.slice(0,(wordNoSymbols.length/2));
+var secondHalf = wordNoSymbols.slice((wordNoSymbols.length/2),wordNoSymbols.length);
+var intermediary = "";
+for(let i = secondHalf.length; i > -1; i--) {
+    intermediary += secondHalf.charAt(i);
+}
+if(intermediary == firstHalf) {
+    console.log("Your word is a Palindrome!")
+}
+
```

Comment:
I understand your logic, it's really good but you are using a complicated way to solve it here, 
If you use the split and reverse methods, you can do it easier ;)